Package LabVIEW Memory Management Tools in a separate package

Karl-G1 opened the pull request:
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-custom-device-development-tools/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Package LabVIEW Memory Management Tools in a separate package

### Why should this Pull Request be merged?

Enable easier distribution for LVMM component without requiring VeriStand be installed.

### What testing has been done?

Verified PR build

buckd commented on control_lvmm:
```diff
@@ -0,0 +1,15 @@
+Package: labview-memory-management-tools-{labview_version}-support
```
The NIPM convention is to have <product>-<version>-support, so
`labview-memory-management-tools-labview-{labview_version}-support`.

buckd replied:
Also, the package name should start with the company.

`ni-labview-memory-management-tools-labview-{labview_version}-support`

buckd commented on build.toml:
```diff
@@ -82,7 +82,14 @@ project = '{network_comm}'
+type = 'nipkg'
+control_file = 'control_lvmm'
+payload_dir = 'lv-memory-manager\Built'
+install_destination = 'ni-paths-LV{labview_version}DIR'
```
Add a newline to the end of `build.toml` per convention.

Karl-G1 commented:
@buckd I corrected the couple fixes you suggested and published the PR for review. 
My biggest open question is if we should 1) conflict with the full dev tools package or 2) exclude LVMM from the full dev tools package.

rtzoeller commented:
> @buckd I corrected the couple fixes you suggested and published the PR for review.
> My biggest open question is if we should 1) conflict with the full dev tools package or 2) exclude LVMM from the full dev tools package.

If we make them conflict, we should also make the dev tools package "provide" the LVMM package name, so it can replace it in a dependency chain. But I would prefer to exclude LVMM from the main package, and not have duplicate binaries between them.

Karl-G1 commented:
I've removed LVMM from the ni-custom-device-development-tools package by removing the copy step from the post-build action VI. I've also added documentation about the package split to the readmes.

rtzoeller commented on README.md:
```diff
@@ -35,20 +35,23 @@ The following dependencies are used by some or all of the included components of
 This repository will regularly release builds of the dependencies using NI Packages per LabVIEW version. Custom device developers and open-source contributors should look for updated dependencies before submitting code to custom device repositories on github.com/ni.
 
+### Packages
+Most of the components are included in a common NI Package (`ni-custom-device-development-tools`). Some components are packaged individually, such as the LabVIEW Memory Manager tools. See the table above for which package contains each component.
```
I feel like this should go above the table. That would also solve the back-referencing, i.e. you could write "See the table below...".

Karl-G1 replied:
Updated to move comment above table.

rtzoeller commented on build.toml:
```diff
@@ -82,7 +82,16 @@ project = '{network_comm}'
+control_file = 'control'
 payload_dir = 'Built'
 install_destination = 'ni-paths-LV{labview_version}DIR'
+package_output_dir = 'Built'
```
You don't need this line.

Karl-G1 replied:
I had just included it for readability. Removed now.

Karl-G1 commented:
Before merging, I noticed that the files for LVMM were not being archived along with the artifacts from the rest of the builds. Talking with @rtzoeller, the easiest solution was to store other builds in Built\Common and LVMM in Built\LMM. This way, we will build packages out of those locations and persist all files. The latest build will test these changes.
